homeassistant/components/imap/config_flow.py

```diff
@@ -128,3 +139,56 @@ async def async_step_reauth_confirm(
             ),
             errors=errors,
         )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> OptionsFlow:
+        """Get the options flow for this handler."""
+        return OptionsFlow(config_entry)
+
+
+class OptionsFlow(config_entries.OptionsFlowWithConfigEntry):
+    """Option flow handler."""
+
+    def validate_key_options_input(
+        self, user_input: dict[str, Any] | None
+    ) -> dict[str, str]:
+        """Validate the user input against other config entries."""
```

Comment:
I think we should allow options flows to use the helper:

https://github.com/home-assistant/core/blob/96225bb287cabd52791300c21ce06a70886f8f3b/homeassistant/config_entries.py#L1518-L1525

Reply:
I have been looking to get access to this validator. Any hints on how to achieve this?

Reply:
I think we need to implement it in the options flow base class again since options flows don't have the domain readily available, needed for the `_async_current_entries` method.

Long term we may want to standardize a bit more around options flows and make the domain of the options flow available to the flow.

Reply:
Maybe you can make it available as part of `OptionsFlowWithConfigEntry` instead of `OptionsFlow`, so you have access to the config_entry?

Reply:
We have the handler which is the config entry id. So we can get the config entry via `self.hass.config_entries`.

Reply:
The config entry is not the problem. It is y

> I think we need to implement it in the options flow base class again since options flows don't have the domain readily available, needed for the `_async_current_entries` method.

That is why I `added validate_key_options_input` to the `OptionFlow` this is just doing the suggested validation, or am I missing something?

Reply:
I think we should have a common helper for options flows too so we don't need to re-implement this in every integration.

We can do it as a follow up though.

Reply:
We need the domain to get all the config entries of the domain. We need the config entry of the option flow to get the domain.

Reply:
Clear